packages/utils/src/Utils.ts

```diff
@@ -246,3 +250,22 @@ export function getAuthorizationList (op: OperationBase): EIP7702Authorization[]
     return (op as OperationRIP7560).authorizationList ?? []
   }
 }
+
+// call entryPoint.getUserOpHash, but use state-override to run it with specific code (e.g. eip-7702 delegate) on the sender's code.
+export async function callGetUserOpHashWithCode (entryPoint: IEntryPoint, userOp: UserOperation): Promise<string> {
+  let stateOverride = null
+  if (userOp.eip7702Auth != null) {
+    const deployedDelegateCode: string = hexConcat([EIP_7702_MARKER_CODE, userOp.eip7702Auth.address])
+    stateOverride = {
+      [userOp.sender]: {
+        code: deployedDelegateCode
+      }
+    }
+  }
+  return await (entryPoint.provider as JsonRpcProvider).send('eth_call', [
```

Comment:
why not entryPoint.callStatic.getUserOpHash ?

Reply:
The problem is that ethers.js for ```getUserOpHash(userOp: PackedUserOperationStruct, overrides?: CallOverrides)``` does not support EIP-7702 tuples at all. This code can be replaced when EIP-7702 support becomes wider.